src/server/scripts/Events/midsummer.cpp

```diff
@@ -210,15 +211,290 @@ class spell_gen_crab_disguise : public AuraScript
+
+        if (!_bunny)
+            _bunny = DoSpawnCreature(NPC_RIBBON_POLE_FIRE_SPIRAL_BUNNY, 0, 0, 0, 0, TEMPSUMMON_MANUAL_DESPAWN, 0);
+    }
+
+    void DoCleanupChecks()
+    {
+        if (_dancerList.empty())
+            return;
+
+        // remove non-dancing players from list
+        std::erase_if(_dancerList, [](Player* dancer)
+            {
+                return !dancer->HasAura(SPELL_RIBBON_POLE_PERIODIC_VISUAL);
+            });
+    }
+
+    void DoFlameCircleChecks()
+    {
+        if (!_ribbonPole)
+            return;
+        if (_dancerList.size() >= THRESHOLD_FLAME_CIRCLE)
+        {
+            // random blue / red circle
+            if (urand(0, 1))
+                _ribbonPole->CastSpell(me, SPELL_FLAME_RING);
+            else
+                _ribbonPole->CastSpell(me, SPELL_FLAME_PATCH);
+        }
+    }
+
+    void DoFireworkChecks()
+    {
+        if (!_bunny)
+            return;
+
+        if (_dancerList.size() >= THRESHOLD_FIREWORK)
+        {
+            _bunny->CastSpell(nullptr, SPELL_RIBBON_POLE_FIREWORK);
+        }
+        if (_dancerList.size() >= THRESHOLD_FIREWORK_3)
+        {
+            _scheduler.Schedule(500ms, [this](TaskContext /*context*/)
+            {
+                if (_bunny)
+                    _bunny->CastSpell(nullptr, SPELL_RIBBON_POLE_FIREWORK);
+            })
+            .Schedule(1s, [this](TaskContext /*context*/)
+            {
+                if (_bunny)
+                    _bunny->CastSpell(nullptr, SPELL_RIBBON_POLE_FIREWORK);
+            });
+        }
+        if (_dancerList.size() >= THRESHOLD_FIREWORK_5)
+        {
+            _scheduler.Schedule(1500ms, [this](TaskContext /*context*/)
+            {
+                if (_bunny)
+                    _bunny->CastSpell(nullptr, SPELL_RIBBON_POLE_FIREWORK);
+            })
+            .Schedule(2s, [this](TaskContext /*context*/)
+            {
+                if (_bunny)
+                    _bunny->CastSpell(nullptr, SPELL_RIBBON_POLE_FIREWORK);
+            });
+        }
+    }
+
+    void DoGroundFlowerChecks()
+    {
+        if (!_bunny)
+            return;
+
+        if (_dancerList.size() >= THRESHOLD_GROUND_FLOWERS)
+        {
+            std::list<Creature*> crList;
+            me->GetCreaturesWithEntryInRange(crList, 20.0f, NPC_GROUND_FLOWER);
+
+            if (crList.size() < MAX_COUNT_GROUND_FLOWERS)
+                _bunny->CastSpell(nullptr, SPELL_RIBBON_POLE_GROUND_FLOWER);
+        }
+    }
+
+    void DoSpewLavaChecks()
+    {
+        if (!_bunny)
+            return;
+
+        if (_dancerList.size() >= THRESHOLD_SPEW_LAVA)
+        {
+            if (!_dancerList.empty())
+            {
+                std::vector<Player*>::iterator begin = _dancerList.begin(), end = _dancerList.end();
+                std::random_device rd;
+                std::shuffle(begin, end, std::default_random_engine{ rd() });
```

Comment:
https://github.com/azerothcore/azerothcore-wotlk/blob/366b8d01fc0c90d97a922893aee03c5010418867/src/common/Utilities/Containers.h#L135

Reply:
Thanks for the hint, will update the code to use utilities